torch/csrc/dynamo/eval_frame.c

```diff
@@ -420,7 +425,6 @@ inline static void set_extra_state(PyCodeObject* code, ExtraState* extra_state)
   //  set_extra_state responsibility to clean it up. It will be deleled during
   //  the reset_code/skip, when the set_extra_state is called with
   //  NULL/SKIP_CODE.
-  destroy_extra_state(code);
   _PyCode_SetExtra((PyObject*)code, extra_index, extra_state);
```

Comment:
I am assuming `_PyCode_SetExtra` calls `destroy_extra_state` if the value is already set?

(I have not confirmed this.)

Reply:
Yes, it does

![image](https://github.com/pytorch/pytorch/assets/13822661/b41431ea-677e-492e-b55c-dd7bff534b68)